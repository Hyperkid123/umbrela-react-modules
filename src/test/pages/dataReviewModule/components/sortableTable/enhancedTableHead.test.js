import React from 'react'
import EnhancedTableHead from '../../../../../pages/dataReviewModule/components/sortableTable/enhancedTableHead';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Enhanced table head component', () => {
  const initialProps = {}
  beforeEach(() => {
    initialProps.onRequestSort = jest.fn();
    initialProps.orderBy = 'name';
    initialProps.order = 'asc';
    initialProps.columnData = [{
      id: 1,
      numeric: false,
    }, {
      id: 33,
      numeric: true
    }]
  });

  it('should render correctly', () => {
    const tree = shallow(
      <EnhancedTableHead {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should call sort', () => {
    const onRequestSort = jest.fn();
    const wrapper = shallow(
      <EnhancedTableHead {...initialProps} onRequestSort={onRequestSort}/>
    )
    wrapper.instance().createSortHandler('foo');
    expect(onRequestSort.mock.calls.length).toEqual(0)
  })
});
