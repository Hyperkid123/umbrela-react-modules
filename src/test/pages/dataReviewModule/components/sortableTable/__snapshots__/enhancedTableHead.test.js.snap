// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enhanced table head component should render correctly 1`] = `
<TableHead
  component="thead"
>
  <WithStyles(TableRow)>
    <WithStyles(TableCell)
      key="1"
      numeric={false}
      sortDirection={false}
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        placement="bottom-start"
        title="Seřadit"
      >
        <WithStyles(TableSortLabel)
          active={false}
          direction="asc"
          onClick={[Function]}
        />
      </WithStyles(Tooltip)>
    </WithStyles(TableCell)>
    <WithStyles(TableCell)
      key="33"
      numeric={true}
      sortDirection={false}
    >
      <WithStyles(Tooltip)
        enterDelay={300}
        placement="bottom-end"
        title="Seřadit"
      >
        <WithStyles(TableSortLabel)
          active={false}
          direction="asc"
          onClick={[Function]}
        />
      </WithStyles(Tooltip)>
    </WithStyles(TableCell)>
  </WithStyles(TableRow)>
</TableHead>
`;
